src/Illuminate/Mail/Message.php

```diff
@@ -228,42 +237,26 @@ protected function createAttachmentFromPath($file)
      */
     public function attachData($data, $name, array $options = [])
     {
-        $attachment = $this->createAttachmentFromData($data, $name);
-
-        return $this->prepAttachment($attachment, $options);
-    }
+        $this->message->attach($data, $name, $options['mime'] ?? null);
 
-    /**
-     * Create a Swift Attachment instance from data.
-     *
-     * @param  string  $data
-     * @param  string  $name
-     * @return \Swift_Attachment
-     */
-    protected function createAttachmentFromData($data, $name)
-    {
-        return new Swift_Attachment($data, $name);
+        return $this;
     }
 
     /**
-     * Embed a file in the message and get the CID.
+     * Embed a file in the message.
      *
      * @param  string  $file
-     * @return string
+     * @return $this
      */
     public function embed($file)
     {
-        if (isset($this->embeddedFiles[$file])) {
-            return $this->embeddedFiles[$file];
-        }
+        $this->message->embedFromPath($file);
 
-        return $this->embeddedFiles[$file] = $this->swift->embed(
-            Swift_Image::fromPath($file)
-        );
+        return $this;
```

Comment:
You could use a custom cid name: https://symfony.com/doc/current/mailer.html#embedding-images
And return this one or let the Developer decide via an argument how the cid should be named.

Maybe defaulting to a random unique one?

Reply:
Don't you need it when generating a link to an image in your email?

Reply:
<img width="783" alt="CleanShot 2021-09-03 at 11 19 09@2x" src="https://user-images.githubusercontent.com/463230/132037402-994e59bf-f2a8-477f-9999-d0825e9e4a73.png">

Reply:
@taylorotwell the line above currently is
```php
$this->message->embedFromPath($file);
```

and this could be changed to:
```php
$this->message->embedFromPath($file, $cid);
```

This cid could then be returned.
Before this cid was generated by Swift. But now Laravel needs to generate the cid.

Reply:
Yeah, I mean we need some story for embedding images obviously. We can't just remove image embedding. Symfony Mailer docs seem to show code embedding them while building the message, manually providing a CID, then referencing that in your HTML:

<img width="767" alt="CleanShot 2021-09-03 at 11 33 36@2x" src="https://user-images.githubusercontent.com/463230/132039101-b97ba6e6-9694-414b-9cb1-25dda9315cbc.png">

Reply:
I pushed a commit that fixes this. Thanks for noting!